creusot-contracts/src/std/num.rs

```diff
@@ -202,6 +202,30 @@ macro_rules! spec_unsized {
                     #[check(ghost)]
                     #[ensures(result == (self != $zero && self & (self - $one) == $zero))]
                     fn is_power_of_two(self) -> bool;
+
+                    #[check(ghost)]
+                    #[ensures(self >> $type::BITS - result == $zero)]
+                    #[ensures((result != $type::BITS) == (self >> ($type::BITS - result - 1u32) == $one))]
+                    #[ensures((result == $type::BITS) == (self == $zero))]
```

Comment:
```suggestion
                    #[ensures((result != $type::BITS) == (self >> ($type::BITS - result - 1u32) == $one))]
                    #[ensures((result == $type::BITS) == (self == $zero))]
```

This clause is redundant (though it's unclear whether SMT solvers will be more comfortable with suc or such clause).